crates/db_views_moderator/src/modlog_combined_view.rs

```diff
@@ -0,0 +1,1405 @@
+      ModlogCombinedView::ModBanFromCommunity(v) => {
+        ("ModBanFromCommunity", v.mod_ban_from_community.id.0)
+      }
+      ModlogCombinedView::ModFeaturePost(v) => ("ModFeaturePost", v.mod_feature_post.id.0),
+      ModlogCombinedView::ModHideCommunity(v) => ("ModHideCommunity", v.mod_hide_community.id.0),
+      ModlogCombinedView::ModLockPost(v) => ("ModLockPost", v.mod_lock_post.id.0),
+      ModlogCombinedView::ModRemoveComment(v) => ("ModRemoveComment", v.mod_remove_comment.id.0),
+      ModlogCombinedView::ModRemoveCommunity(v) => {
+        ("ModRemoveCommunity", v.mod_remove_community.id.0)
+      }
+      ModlogCombinedView::ModRemovePost(v) => ("ModRemovePost", v.mod_remove_post.id.0),
+      ModlogCombinedView::ModTransferCommunity(v) => {
+        ("ModTransferCommunity", v.mod_transfer_community.id.0)
+      }
+    };
+    // hex encoding to prevent ossification
+    ModlogCombinedPaginationCursor(format!("{prefix}{id:x}"))
+  }
+
+  pub async fn read(&self, pool: &mut DbPool<'_>) -> Result<PaginationCursorData, Error> {
+    let err_msg = || Error::QueryBuilderError("Could not parse pagination token".into());
+    let mut query = modlog_combined::table
+      .select(ModlogCombined::as_select())
+      .into_boxed();
+    let (prefix, id_str) = self.0.split_at_checked(1).ok_or_else(err_msg)?;
+    let id = i32::from_str_radix(id_str, 16).map_err(|_err| err_msg())?;
+    query = match prefix {
+      "AdminAllowInstance" => query.filter(modlog_combined::admin_allow_instance_id.eq(id)),
+      "AdminBlockInstance" => query.filter(modlog_combined::admin_block_instance_id.eq(id)),
+      "AdminPurgeComment" => query.filter(modlog_combined::admin_purge_comment_id.eq(id)),
+      "AdminPurgeCommunity" => query.filter(modlog_combined::admin_purge_community_id.eq(id)),
+      "AdminPurgePerson" => query.filter(modlog_combined::admin_purge_person_id.eq(id)),
+      "AdminPurgePost" => query.filter(modlog_combined::admin_purge_post_id.eq(id)),
+      "ModAdd" => query.filter(modlog_combined::mod_add_id.eq(id)),
+      "ModAddCommunity" => query.filter(modlog_combined::mod_add_community_id.eq(id)),
+      "ModBan" => query.filter(modlog_combined::mod_ban_id.eq(id)),
+      "ModBanFromCommunity" => query.filter(modlog_combined::mod_ban_from_community_id.eq(id)),
+      "ModFeaturePost" => query.filter(modlog_combined::mod_feature_post_id.eq(id)),
+      "ModHideCommunity" => query.filter(modlog_combined::mod_hide_community_id.eq(id)),
+      "ModLockPost" => query.filter(modlog_combined::mod_lock_post_id.eq(id)),
+      "ModRemoveComment" => query.filter(modlog_combined::mod_remove_comment_id.eq(id)),
+      "ModRemoveCommunity" => query.filter(modlog_combined::mod_remove_community_id.eq(id)),
+      "ModRemovePost" => query.filter(modlog_combined::mod_remove_post_id.eq(id)),
+      "ModTransferCommunity" => query.filter(modlog_combined::mod_transfer_community_id.eq(id)),
```

Comment:
This could also potentially work with strum to convert string into enum variant with default value 0 as id. That way you dont need to type out strings for the match which could have typos.

https://docs.rs/strum/latest/strum/derive.EnumString.html